### PR TITLE
Fixes #12432 - Changed roles model so admin could see/attach all permissions

### PR DESCRIPTION
**Pull request by alongoldboim**

### CODE REVIEW

**Comment by domcleal** [comment]
As per IRC, please can you add a unit test to cover this scope?

**Comment by alongoldboim** [comment]
@domcleal test added.

File: test/unit/role_test.rb [hunk 1]
```diff
@@ -133,6 +133,16 @@ class RoleTest < ActiveSupport::TestCase
       it { subject.must_include(first) }
       it { subject.wont_include(second) }
     end
+
+    context "when current user is admin for_current_user should return all roles" do
+      before do
+        User.current = users(:admin)
+      end
+      tmp_role = Role.create(:name => 'First')
```
Comment (dLobatog) [inline test/unit/role_test.rb]: @alongoldboim What is this `tmp_role` for? It's not being tested anywhere, if you remove it, the test will still pass.

Maybe you want to test it like:

``` ruby
Role.expects(:where).with('0 = 0')
```

to ensure it tests the change you made, no need to actually check what's inside the db call. Does that sound correct?

File: test/unit/role_test.rb [hunk 2]
```diff
@@ -133,6 +133,16 @@ class RoleTest < ActiveSupport::TestCase
...
+      tmp_role.save
+      it { assert_equal Role.for_current_user, Role.all }
```
Comment (domcleal) [inline test/unit/role_test.rb]: Not sure you need the spec-style `it { .. }` here, just assert?

File: test/unit/role_test.rb (hunk 1 above)
Comment (alongoldboim) [inline test/unit/role_test.rb]: @dLobatog sounds right :), changing and pushing.

**Comment by alongoldboim** [comment]
@domcleal test added.

File: test/unit/role_test.rb [hunk 3]
```diff
@@ -133,6 +133,16 @@ class RoleTest < ActiveSupport::TestCase
       it { subject.must_include(first) }
       it { subject.wont_include(second) }
     end
+
+    context "when current user is admin for_current_user should return all roles" do
+      before do
+        User.current = users(:admin)
+      end
+      it "Admin user should query Role model with no restrictions" do
```
Comment (dLobatog) [inline test/unit/role_test.rb:141]: I'll change the 'it' to 'test'. Not sure why rubocop is not complaning about the missing empty line between the setup block and this.  @alongoldboim don't worry I'll change that on merge.

**Comment by dLobatog** [comment]
Merged as b2a959ec283ae96ec5d33fbf5f484e3f3fa07657, thanks @alongoldboim !

**Pull request closed without merging**